packages/eslint-plugin/tests/rules/no-unsafe-assignment.test.ts

```diff
@@ -195,7 +196,50 @@ class Foo {
       `,
       errors: [{ messageId: 'anyAssignment' }],
     },
+    {
+      code: `
+const [x] = spooky;
+      `,
+      errors: [{ messageId: 'anyAssignment' }],
```

Comment:
[Testing] These first two added test cases should have the `data` too. Every test added in this PR is something where the data having `error` or `any` is important enough to test, I'd think.